### PR TITLE
Add LeNet Implementation in PyTorch

### PR DESCRIPTION
**Pull request by ishandutta0098**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: computer_vision/lenet_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+Paper: http://vision.stanford.edu/cs598_spring07/papers/Lecun98.pdf
+"""
+
+import torch
+import torch.nn as nn
+
+
+class LeNet(nn.Module):
+    def __init__(self):
```
[the remark this answers is not in the document]
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Done

File: computer_vision/lenet_pytorch.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            out_channels=120,
+            kernel_size=(5, 5),
+            stride=(1, 1),
+            padding=(0, 0),
+        )
+
+        self.linear1 = nn.Linear(120, 84)
+        self.linear2 = nn.Linear(84, 10)
+
+    def forward(self, x):
```
[the remark this answers is not in the document]
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Done

File: computer_vision/lenet_pytorch.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Done

File: computer_vision/lenet_pytorch.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
```
[the remark this answers is not in the document]
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Done

File: computer_vision/lenet_pytorch.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Paper: http://vision.stanford.edu/cs598_spring07/papers/Lecun98.pdf
+"""
+
+import numpy
+import torch
+import torch.nn as nn
+
+
+class LeNet(nn.Module):
+    def __init__(self):
```
[the remark this answers is not in the document]
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Done

File: computer_vision/lenet_pytorch.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            out_channels=120,
+            kernel_size=(5, 5),
+            stride=(1, 1),
+            padding=(0, 0),
+        )
+
+        self.linear1 = nn.Linear(120, 84)
+        self.linear2 = nn.Linear(84, 10)
+
+    def forward(self, x: numpy.ndarray) -> numpy.ndarray:
```
[the remark this answers is not in the document]
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Done

[the remark this answers is not in the document]
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Done

File: computer_vision/lenet_pytorch.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
```
[the remark this answers is not in the document]
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Done

**Comment by ishandutta0098** [comment]
Hi, does this require any further changes?

**Review by cclauss — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 7]
```diff
@@ -15,6 +15,7 @@ statsmodels
 sympy
 tensorflow
 texttable
+torch
```
Comment (cclauss) [inline requirements.txt:18]: https://github.com/pytorch/pytorch/issues/86566

Comment (ishandutta0098) [reply requirements.txt:18]: Do I need to do anything about this?

File: computer_vision/lenet_pytorch.py [hunk 8]
```diff
@@ -54,20 +54,28 @@ def forward(self, image_array: numpy.ndarray) -> numpy.ndarray:
...
     True
 
     """
-    random_image = torch.randn(64, 1, 32, 32)
-    model = LeNet()
-    output = model(random_image)
+    try: 
+        model = LeNet()
+        output = model(image_tensor)
+    except:
```
Comment (cclauss) [inline computer_vision/lenet_pytorch.py:71]: https://realpython.com/the-most-diabolical-python-antipattern

computer_vision/lenet_pytorch.py:71:5: B001 Do not use bare `except:`, it also catches unexpected events like memory errors, interrupts, system exit, and so on.

computer_vision/lenet_pytorch.py:71:5: E722 do not use bare 'except'

File: requirements.txt (hunk 7 above)
Comment (cclauss) [reply requirements.txt:18]: The only thing we can do is wait until torch is compatible with the current version of Python.

File: computer_vision/lenet_pytorch.py (hunk 8 above)
Comment (ishandutta0098) [reply computer_vision/lenet_pytorch.py]: Corrected

File: requirements.txt (hunk 7 above)
Comment (ishandutta0098) [reply requirements.txt:18]: Okay, after it is compatible this PR will be automatically merged?

**Comment by ishandutta0098** [comment]
@cclauss can this be merged now?

**Comment by cclauss** [comment]
1. https://github.com/pytorch/pytorch/issues/86566
2. Does the test prove that the algorithm is delivering the accurate result?

**Review by cclauss — APPROVED** [review]
RSLGMT

**Pull request merged**